src/main/java/com/avairebot/imagegen/RankBackgroundHandler.java

```diff
@@ -0,0 +1,313 @@
+     */
+    @Nullable
+    public RankBackground fromName(@Nonnull String name) {
+        for (RankBackground background : backgrounds) {
+            if (background.getName().equalsIgnoreCase(name)) {
+                return background;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Gets the rank background with the given ID, if no rank backgrounds were
+     * found with the given ID, {@code NULL} will be returned instead.
+     *
+     * @param backgroundId The ID of the rank background that should be returned.
+     * @return Possibly -null, the rank background with a matching ID, or {@code NULL}.
+     */
+    @Nullable
+    public RankBackground fromId(int backgroundId) {
+        for (RankBackground background : backgrounds) {
+            if (background.getId() == backgroundId) {
+                return background;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Initializes the rank background containers, loading all the rank
+     * background information into memory, this method should only
+     * be called once during the startup of the bot.
+     */
+    public void start() {
+        Map<String, Integer> unsortedNamesToCost = new HashMap<>();
+
+        try {
+            for (RankBackground type : getResourceFiles()) {
+                unsortedNamesToCost.put(type.getName(), type.getCost());
+                backgrounds.add(type);
+            }
+        } catch (IOException e) {
+            System.out.printf("Invalid cache type given: %s", e.getMessage());
+            System.exit(ExitCodes.EXIT_CODE_ERROR);
+        }
+
+        unsortedNamesToCost.entrySet().stream()
+            .sorted(Map.Entry.comparingByValue())
+            .forEach(entry -> namesToCost.put(entry.getKey(), entry.getValue()));
+    }
+
+    /*
+     * Copies every file in the backgrounds resource if it does not already
+     * exist in the backgrounds folder of the current AvaIre instance.
+     * Then updates the background index at the root of the current running
+     * directory.
+     *
+     * If the file could not be copied or any possible errors occur,
+     * the bot will print the exception to the string and subsequently shut down.
+     */
+    private void copyBackgroundsFromJarToFolder() {
+        try {
+            List<String> files = ResourceLoaderUtil.getFiles(RankBackgroundHandler.class, "backgrounds");
+            for (String file : files) {
+                File actualFile = new File("backgrounds/" + file);
+                InputStream inputStream = RankBackgroundHandler.class.getClassLoader().getResourceAsStream("backgrounds/" + file);
+                if (!actualFile.exists()) {
+                    Files.copy(inputStream, Paths.get("backgrounds/" + file), StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+            writeToIndex(files);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
You should use the slf4j logger for logging the output so the exception can both have a message attached to it, and the exception can be caught by [Sentry](https://sentry.io/), which can help track down the error later, and figure out what exactly went wrong.
The same goes for the other exceptions thrown in this class that just uses the `System.out.print` for the message.